### DISCUSSION TITLE
How to stop underlying process on gradle task interrupt?

### DISCUSSION THREAD

**Discussion by AuroraLS3** [node-gradle/gradle-node-plugin#260] [Q&A] [answered]
Hello!

I am trying to setup an environment agnostic way to start the react dev server through gradle-node-plugin.
(While I agree that this may be questionable since gradle is a build tool, it is the easiest way to have all the build tools and dependencies installed on a fresh installation through gradle-wrapper)

### Problem

Stopping the `react-scripts start` process usually happens through keyboard interrupt. Unfortunately when you keyboard interrupt a running YarnTask it does not kill the underlying node process, so it is left hanging without direct way to kill it

How should I approach this?

**Comment by deepy** [comment]
This is unfortunately something that should be handled by Gradle but isn't, issue https://github.com/node-gradle/gradle-node-plugin/issues/65 has the background for it and https://github.com/node-gradle/gradle-node-plugin/issues/65#issuecomment-872062591 has a possible workaround.

I haven't tried the workaround personally but it looks sound and is worth a try

**Reply by AuroraLS3** [reply]
Ah bummer, I ran across a similar closed issue so I thought it had been solved. Thanks for the workaround, I'll see if it works :)

**Reply by AuroraLS3** [reply, marked as answer]
Setting `--no-daemon` in the script seems to kill the process afterwards :)
